### ISSUE TITLE
[Feature Request] Suppress empty repeated panels

### ISSUE THREAD

**Issue by vitorboschi** [grafana/grafana#4470]
Use case: 
Server A have eth0, eth1 and server B have eth0 and ppp0.

I create a templated variable $netInterface and use the query "tag_values(os.net.bytes,iface)" in it, so it will get the possible values "eth0, eth1, ppp0". 
When using this variable with the repeated panel feature, I'll get 3 panels (one for each value), but in server A, the ppp0 panel will be empty, and on server B eth1 panel will be empty. If I could configure Grafana to suppress the empty ones (just like I can suppress empty series' legend), the dashboard would be much easier to interpret.

**Comment by vitorboschi** [comment]
I've been thinking about this feature and came to conclusion that it may be useful in other contexts other than repeated panels, so I though about 3 possibilities:

1. checkbox next to the repeated panel options
2. checkbox next to the "Transparent" checkbox. 
3. dashboard wide option, so it would be in the settings, along with the "shared crosshair", "Editable"... options

1 and 2 have a problem that once enabled, you could not access the suppressed panel, so another flag to force displaying of suppressed panels. 3 would mean it cannot be configured for each panel.

I guess a combination of 1 and 3 would be the best solution, but I'd like to hear your opinions as well

**Comment by bergquist** [comment]
I like the idea of suppressing empty panels. But I'm not sure about adding two new check boxes for repeat panel. Is there any other way to settling for less options?

**Comment by vitorboschi** [comment]
The only alternative way is to have just a dashboard wide option to hide empty panels, but I'm open to suggestions.
If it's just on the panels, how do the user un-hide a previously hidden panel?

**Comment by DEvil0000** [comment]
I think #5171 would also solve this and is more flexible (hide/show repeated panels based on threshold).

**Comment by tfylling** [comment]
I would love this feature on both row and template repeats, alternatively on a dashboard level if that's easier / more sensible / less complexity.
A few examples where it would be extremely useful is when selecting multiple network interfaces from multiple hosts, where some are using ethX and others are using emX, have different vlans etc. A dashboard where half the panels are empty has no value, and there is no good way to work around the issue.

Another example is statistics from multiple databases with different names across hosts.

**Comment by 0111sandesh** [comment]
Any update on this feature? I am also facing similar issues where I have templated dashboards with some graphs without datapoints and SingleStat panels with NA values.

**Comment by Corbyn** [comment]
This feature is really a must have!
We use Grafana as frontend for all our monitoring performance data. And the drawback of host independent time series like you have them in InfluxDB is, that at the moment it's not possible to show only the relevant graphs in one dashboard.
E. g. all disk usage data is stored in one measurement, but you want to display only the disk partitions that exist on any given host, then you need the possibility to hide empty rows/panels otherwise you would have a list of all partitions that exist in your complete server landscape...

And if I understand it right, it should be possible to make the loading of repeated rows/panels dynamic, just as it is with template variable values where you can choose when to reload them (Never, On Dashboard Load...)

**Comment by alekseyp** [comment]
👍

**Comment by n3k232** [comment]
It would be nice to have the Function to Hide empty Panels, that would help me alot.

**Comment by cdlwjing** [comment]
yeah, I also would like to use this function, could it be publish to public version?

**Comment by rbrackma** [comment]
Yes, we are also waiting for that!

**Comment by eohan01** [comment]
+1

**Comment by jan-berger** [comment]
I would also like this.

**Comment by rojobull** [comment]
Yes. this would be great!

**Comment by theobolo** [comment]
+1

**Comment by m12o** [comment]
+1

**Comment by jeremysherriff** [comment]
+1

**Comment by polak785** [comment]
+1

**Comment by kccheung** [comment]
+1

**Comment by daixiang0** [comment]
Now any update?
I am really waiting for this and it would be a cool feature.

**Comment by viberan** [comment]
+1

**Comment by Armadill0** [comment]
I would really appreciate this feature! :+1: 

For me an implementation like minimizing the panels which contain no data points would be sufficient. This could be done on the dashboard level.

Minimizing could be done like that:
- only the top panel with the label and the dropdown menu could be shown
- add a "no data points" label e.g. on the right side of the top bar
- add a maximize button, e.g. a "+" symbol also on the far right of the top bar or "maximize" menu entry in the panel's dropdown menu, to be able to show the panel again for debugging purposes and so on

**Comment by ultrabug** [comment]
+1 :+1:

**Comment by tvsaru** [comment]
👍

**Comment by rscruz7** [comment]
+1

**Comment by rowco** [comment]
+1

**Comment by faxm0dem** [comment]
please stop posting comments containing only +1
this is counter-productive.

_There is a thumbs-up icon in the original issue you can click on for that!_

**Comment by maxlehot1234** [comment]
Any update on this, since a lot of people want this feature to be implemented since 3 years now ;)

Hide no data points will be amazing to keep a beautiful interface without empty graph for monitoring reasons.

Thanks

**Comment by dguendisch** [comment]
Seeing the bunch of PRs trying to make that configurable (but all of them closed mostly on UX reasons), makes me wonder on another option:
if the grafana maintainers do not want to introduce additional UX elements, why isn't suppressing empty repeated panels the default (with no option to activate it at all)?
I mean what's the purpose of dynamic repeating panels to show some with "No data points"?
The information gain is pretty much zero.
So I would opt to never show these empty panels and I would never need a config option to show them.

(Btw. dynamically charts in Kibana react to your filters and timepicker, they never show empty charts and have no option to do so.)

**Comment by torkelo** [comment]
Kibana has nothing like the panel repeat option.

We are introducing another form of repeat with our new react panel (gauge and bar gauge) and upcoming new singlestat and graph will all support repeating based on time series the  query returns instead of a variable (repeat with the panel bounds) so will be much simpler to setup and more perform any as you can repeat with only one query returning many series. The reason we do not hide empty panels is that the panel needs to exist and take up space as the old panel architecture is also handling the query execution, and we do not want invisible panels issuing queries

**Comment by dguendisch** [comment]
Thanks for the info, sounds promising what you plan!
> Kibana has nothing like the panel repeat option.
It does!
Say I have a simple histogramm chart on duration averages:

**Comment by dguendisch** [comment]
Thanks for the info, sounds like promising improvements you plan!
> Kibana has nothing like the panel repeat option.

It does! Say I have a simple time-based chart drawing some duration averages.
I can create a bucket of type "Split chart" and decide to e.g. have n charts for all available values of a label(field), resulting in this:
![image](https://user-images.githubusercontent.com/30311254/57531075-f22e6380-7338-11e9-9dcc-7d665f2b7f67.png)
I can decide whether I want the repeat to happen horizontally or vertically and so on.

**Comment by arianofaoms** [comment]
Big PLUS 1 here :-)

**Comment by krisavi** [comment]
https://community.grafana.com/t/hiding-a-row-panel/1788/4 According to that post they have stated that this will not happen.
I was hoping to have it hidden as well, since measuring different services performance on different hosts. Some hosts have services x,y, some have a,x and some only z. The whole screen has a lot of "No data points" and looks a bit empty due to having 6 different services and handful of hosts but services not on all the machines.

Maybe one approach would be to somehow specify with the query how many panels it should have to format correctly, but as I saw that one answer then it is highly likely not happening.

So, maybe it is best that you should close the issue and give more official answer in here on why it is not happening. From the conversation it seems some have implemented something in PRs but it was always rejected due to UX questions and adding some checkboxes. From those rejections it seems to be more of GrafanaLabs team deciding they do not want to have feature like that. Would be nice to let users know at least of decision like that.

**Comment by draeron** [comment]
one way of solving this would be to support variable / query as a source for repeated panels / group

**Comment by baur** [comment]
one way of solving this would be to just support fire RowID change event /for Row Repeat ID/ when Row is rendering in order to load another PanelID list /for Panel Repeat ID/ and set query like this: `select * from table where rowid=$RowID and panelid=$PanelID ` in this case we must also support hide mode of variables /block interacting/ or not ...

**Comment by phemmer** [comment]
On the subject of hiding repeating panels which use query variables, and why this hasn't been implemented: I understand that hiding repeating panels for which there are no variable values poses a difficulty as then you don't have a panel you can edit.

Would it be possible to solve this by either:
1. Add an option to automatically collapse rows which all the repeated panels have no values? Then the user could just manually expand the row to see & edit it.
2. Add an option to the row to hide repeated panels which have no values? Then the user could toggle the option to see & edit the panel.

Note that I'm not referring to the case where the panel query executes and comes back with no data (the case this issue was opened for), but instead when a variable query executes and comes back with no values (this issue is being used as a duplicate for those cases).

**Comment by 030** [comment]
@torkelo Could you indicate what the state is of this issue and what should be done to implement it?

**Comment by RamonaStefan** [comment]
+1

**Comment by w4rc0n** [comment]
Why has this feature not been added? It's overwhelmingly clear that a ton of people want it, and it can't possibly be that hard.

**Comment by shaunroselt** [comment]
+1

**Comment by JBrVJxsc** [comment]
Not sure why it is so hard to implement this... the pandemic's vaccine is out now

**Comment by jkrauska** [comment]
Repeating here for posterity:
From https://github.com/grafana/grafana/issues/29646

"""
But we won't add conditional logic like this that hides panel (makes them impossible find / edit). The solution for this use case will be new scripted / programmatically defined dashboard that we hope to introduce next year
"""

**Comment by bugsyb** [comment]
Found something interesting but have no clue how it is done on:
- https://grafana.com/grafana/dashboards/8168 - that it nicely hides VMs without any data withing the time range (requires full page reload, not just range update)
- https://grafana.com/grafana/dashboards/5763 - unfortunately shows always all containers (including those without any data in the range).

Those of you more proficient in Grafana will certainly be able to spot the difference. Is it maybe "variables" use? I have no clue and hope someone more smart than me can figure it out for the benefit of us all.

**Comment by Gregorini-Enrico** [comment]
Is there a solution for this issue?

**Comment by TryTryAgain** [comment]
I'd even be open for a Greasemonkey/similar solution if anyone has come up with anything non-native as well...

**Comment by tfylling** [comment]
So I gather one argument against this is that it should be possible to right click the panel to edit it? Also I believe you are not huge fans of options and switches? So then how about hiding panels with no data, and by extension possibly rows with only no data panels, if the dashboard is in read-only mode and show them if it's in edit mode? Possibly with a toggle switch in the panel options (yes, I know that'll still count as an option) wether to hide in read-only mode or not, as we may have some panels that we do want to show, even if they're empty.

Or is it a huge thing that edit and read-only mode should have identical layout? To me it really doesn't matter, but if it does there's no way to get around the first argument without a 'show empty or hidden panels' toggle.

**Comment by grandich** [comment]
This feature would be great for generic or "single pane of glass"-type of dashboards that consume a variety of metric families some of which can be missing depending on the instance / job / datasource.

**Comment by mrtypn** [comment]
I would love if Grafana support supports this feature.

**Comment by juliencorsini** [comment]
+1

**Comment by cybool** [comment]
Hi @torkelo, last year you wrote:
From #29646 (Dec 2020)
`    But we won't add conditional logic like this that hides panel (makes them impossible find / edit). The solution for this use case will be new scripted / programmatically defined dashboard that we hope to introduce next year`

Is it somehow planned for 8.3 or 8.4?

**Comment by Berkmann18** [comment]
This seems like a must and those saying that implementing a checkbox/toggle to hide empty panels (thus solving this issue) would degrade the UX don't realise that the lack of this functionality **is** degrading the UX.
And it's been a year since what @cybool quoted.

**Comment by torkelo** [comment]
The problem introducing this has nothing to do with introducing a ux checkbox, but how queries are executed, dashboard layouts and layout reflows. Hiding empty panels would be easy but would leave lots of blank space , assume that is not what users want. Then having dashboard layout reflow as queries returns is not trivial

**Comment by torkelo** [comment]
And another is how to edit a panel whose queries return empty and is hence hidden

**Comment by grandich** [comment]
I naively see it's implementation as the inverse or a special case of Repeat Panels or Rows (by variable), repeating zero times. And not necessarily leaving blank space, but may be a small marker and a toggle for viewing them, like when text editors toggle 'visible white space/tabs'.

**Comment by amir-am** [comment]
+1

**Comment by YunaBraska** [comment]
+1 I also would like to add a function to show only panel of interest. It can be a filter, sorting or hiding panels without data or with non critical data. This would bring up much more overview. I understand that hiding is not an option as you can't edit or find your panel but there are way more solutions like the ordering of panels...

**Comment by stefankaufmann** [comment]
+1 small feature, but increases usability of dashboards. Especially for dashboards which are controlled by variables. 
Our use case is a dashboard for server monitoring. If you select a Linux server from the variable dropdown, we are also interested in docker statistics. If it is a windows server, we do not want to see all the empty docker stuff there.

**Comment by danielelopez1** [comment]
+1

**Comment by youssefbenfarhat** [comment]
Any news about this feature ?

**Comment by shadowking001** [comment]
It would be a great addition, I confirm !

**Comment by KornKalle** [comment]
Same here, would love to see this

**Comment by camathieu** [comment]
I support this 100%, we can't build flexible dashboards without this.

Things that lacks show/hide capabilities based on arbitrary conditions (no only no data) :
- Dashboard rows
- Dashboard panels
- Panel queries

**Comment by Franck31** [comment]
Any news about this? 

This will be very helpful

**Comment by denniseffing** [comment]
I was just searching the web how to do this because I thought "this **must** be possible, right?". Welp...
This feature would be much appreciated!

**Comment by itsgodzilla** [comment]
Nice feature and just cannot believe it isn't been finished since these long time. Almost can't bear to join the Grafana team and do it myself ...

**Comment by Schmidt-Heinrich** [comment]
We desperately need this on projects that we have. Without this we need to create multiple copies of pages and use a menu system to present the correct pages to the correct individuals. Everyone uses anonymous access.

**Comment by JaimeAvalos** [comment]
another request here

**Comment by agupta-ionos** [comment]
@torkelo are there any plans of introducing this feature in your upcoming Grafana release? Since we are also requiring this feature on our dashboards.

**Comment by torkelo** [comment]
@agupta-ionos yes, in the new dashboard architecture this should be easier to implement https://github.com/grafana/grafana/issues/52484

**Comment by agupta-ionos** [comment]
> @agupta-ionos yes, in the new dashboard architecture this should be easier to implement #52484

Thanks @torkelo for your quick reply, Thats a good thing to know, I have now marked myself as a watcher of that EPIC.

**Comment by simo222** [comment]
Hi

Instead of hiding it, you can just minimize it. 
Then it will take minimaal space on the dashboard
Or move it to the bottom.
I think you should always put an icon where the panel is hidden. With the icon you can always edit again.

**Comment by TryTryAgain** [comment]
> Hi
> 
> Instead of hiding it, you can just minimize it. 
> Then it will take minimaal space on the dashboard
> Or move it to the bottom.
> I think you should always put an icon where the panel is hidden. With the icon you can always edit again.

Not really helpful when you have 300 items that are empty and only 14 with data and the fact that the data is variable. Hiding is not an option.

**Comment by barebu** [comment]
+1

**Comment by MrStefanGit** [comment]
Any updates on hiding repeated panels that have no data?

**Comment by circletmate** [comment]
+1 to this issue, I think it would be useful. the function that I need is rather described in the task [#29172](https://github.com/grafana/grafana/issues/29712), but the solution of this task will solve my problems.

**Comment by agupta-ionos** [comment]
> > @agupta-ionos yes, in the new dashboard architecture this should be easier to implement #52484
> 
> Thanks @torkelo for your quick reply, Thats a good thing to know, I have now marked myself as a watcher of that EPIC.

Hi @torkelo,

When is the EPIC getting introduced in the production?

**Comment by graemechristie** [comment]
Now that #52484 is complete - can we continue the conversation on how we might implement the hiding of panels with no data ?

**Comment by koky75** [comment]
+1

**Comment by limbail** [comment]
+1

**Comment by fmad** [comment]
We have about 50 services per each of 10 namespaces in an AKS cluster - was trying to show a repeating row per namespace (that is working) and then a panel per service but for this particular metric, most will not have values in the selected interval, so there's quite a lot of empty panels (we may have as few as 1 or 2 per namespace), which can vary depending on the time range selected... That's 500 panels displayed, out of which only 20 have data and the remaining 480 are empty, which means we have to do a lot of scrolling up and down for what's mostly empty panels. :(

This has been open for 7 years now in a couple weeks, so I don't know if I even dare asking when these new more flexible dashboards from #52484 will become available? And will we finally have the option to hide empty panels?

**Comment by jdesnoes** [comment]
Hello, would be a nice feature that I am requesting too..!
In my specific case, it would avoid to have half of the panels taking (half of the) dashboard's space for nothing...

**Comment by pieter2501** [comment]
Trying to implement a search function that shows/hides rows based on a dashboard variable. Trying to work around it now using some CSS, but to not much avail.

**Comment by gribaaaaaaaa** [comment]
I think it would be very useful. 

In some cases, e.g. with repeated panels in repeated rows, that based on same variables, it's necessary. Othervise there will be a lot of empty panels.

**Comment by seanosteen** [comment]
Ability to hide empty panels or other conditional would be a welcome addition!

**Comment by dserra3** [comment]
+1 to this issue. It would be very usefull to hide empty pannels or using a condition.

**Comment by ekovacs** [comment]
maybe on this issue's 10th anniversary they make it happen...
:-(

**Comment by jdix531** [comment]
I like to have a dashboard where most panels apply to most deployments. Not everything applies to each though (some panels are for server-level metrics but the app is on shared hardware for instance). I'd like to hide ones that don't have results to help people narrow down their searching.

**Comment by michalkkkd** [comment]
Any news?

**Comment by martinamoro** [comment]
+1

**Comment by chaoukibenfradj** [comment]
+1

**Comment by paulwer** [comment]
+1

**Comment by YustinaKvr** [comment]
Hello Grafana Team!

I've noticed that people around the world have been requesting a specific feature for the past seven years. Throughout this time, your customers have had to rely on various workarounds, each with varying levels of ugliness. We truly believe in your ability to implement this feature and would greatly appreciate your attention to this matter. Thank you for considering our request :)

**Comment by epeters-jrmngndr** [comment]
Maybe a bug bounty could be added to this issue

**Comment by psi-4ward** [comment]
+1 :p

**Comment by majiajun7** [comment]
+1

**Comment by rasti-hasan** [comment]
+1

**Comment by dschoeneborn** [comment]
+1

**Comment by JeKa0111** [comment]
+1 Is there really no update on this for the past 6yrs? 😱

**Comment by gileslauw** [comment]
+1

**Comment by rfmoz** [comment]
+1

**Comment by brettdaman** [comment]
+1

**Comment by JellevdK** [comment]
+1

**Comment by darmach** [comment]
+1

**Comment by kmcrawford** [comment]
+1

**Comment by Codelica** [comment]
**BEFORE you leave a +1 comment on this issue, READ:**

You're about to make a mistake!  All leaving a +1 message does is send a notification email to the (currently 98 people) subscribed to this issue with literally nothing insightful.  So ask yourself what you are trying to achieve:

Do you want to show your support for this issue?  then simply hit one of the response icons on the original post at the top:


<img width="257" alt="response-icons" src="https://github.com/grafana/grafana/assets/386101/1b814098-cff9-4be0-a2ae-8a2b5dd40245">


Do you want to subscribe to updates to find out if this issue will EVER progress?  There is a subscribe button in the right column near the top:


<img width="342" alt="subscribe-button" src="https://github.com/grafana/grafana/assets/386101/623bc660-76d4-4b4c-a781-dbba1a34c897">


So PLEASE, stop with the +1 comments.   Thanks.

**Comment by chenteddybear** [comment]
+1

**Comment by stolsvik** [comment]
+1

**Comment by Dentrax** [comment]
Almost 7 years passed since this initial proposal, I'm curious if this will be prioritized anytime soon or not. A follow-up clarification from Grafana Team would be really helpful to people who need this feature.

**Comment by Huruspeaks** [comment]
Is there any update on this feature please. I want variables to be displayed based on panel during editing the panel nd based on the variables we chose, the panels will be displayed during overall dashboard view.

**Comment by xmoov** [comment]
Wow, welcome to the party I guess. I'll just +1 👍🏽  this and see myself out..

**Comment by yzhang1991** [comment]
+1

**Comment by DmEye** [comment]
I think it is wrong way to hide empty/no data panels to solve the root problem. Because, "no data" panels bring useful information: absence of data. 
The better way is to modify the "Repeat panel's option by variable" option. The modification should implements an ability to repeat panels by query as it is done in dashboard variables.
For example, we have a metric with defined label values:
_bytes_in{server="A", interface="eth0"}_,
_bytes_in{server="A", interface="eth1"}_,
_bytes_in{server="B", interface="eth0"}_,
_bytes_in{server="B", interface="ppp0"}_.
We want to show Row per server, where each Row contains a panel per interface. So, we would set the **$server** varible in the Variables section of a dashboard, create Row with "Repeat for **$server**" option, and add a panel with "Repeat for query _'label_values(bytes_in{server="**$server**", interface})'_" option. In that way, a panel knows which server is and which interfaces it has.

**Comment by vitalii-korotun** [comment]
perhaps, introducing to all panels the simple boolean field which defines either to render panel or not, should be just as enough, if the existing feature "overrides" could be applied to it of course. 
Then people already would do many useful things with it.

.. it just seems like requesting more than that, can take another decade.

**Comment by yatinsharma-swiggy** [comment]
We have use case where we are displaying ec2 and Kubernetes metrics on same panel. Since we have different metrics for different Computes and they have different data source, we woul love to hide the panel that has no data.

**Comment by AHeinlein** [comment]
> I think it is wrong way to hide empty/no data panels to solve the root problem. Because, "no data" panels bring useful information: absence of data. The better way is to modify the "Repeat panel's option by variable" option. The modification should implements an ability to repeat panels by query as it is done in dashboard variables. For example, we have a metric with defined label values: _bytes_in{server="A", interface="eth0"}_, _bytes_in{server="A", interface="eth1"}_, _bytes_in{server="B", interface="eth0"}_, _bytes_in{server="B", interface="ppp0"}_. We want to show Row per server, where each Row contains a panel per interface. So, we would set the **$server** varible in the Variables section of a dashboard, create Row with "Repeat for **$server**" option, and add a panel with "Repeat for query _'label_values(bytes_in{server="**$server**", interface})'_" option. In that way, a panel knows which server is and which interfaces it has.

That doesn't cover all cases - in your example either a server has an interface "ppp0" or it doesn't - that's static. There are, however, usecases where data exists for a given query, just not in the requested time range, and then we want to hide it.

I am currently trying to setup grafana for monitoring linux client pcs. Contrary to servers, clients are not always online and that is just normal. Some of our clients aren't being used for weeks. But currently panels repeating over all clients always show empty for all clients for which there is no data in the requested time range, and I would like to get rid of that.

**Comment by moophat** [comment]
Why is this being removed to Backlog, it's something that people has been voting for for ages

**Comment by ndjhartman** [comment]
+1 :)

**Comment by hiepqdinh** [comment]
A simple way to support this is to allow a variable with no value (or an empty list of values), that way the panels that need to be hidden can just be a repeat of a custom variable which would return an empty list when the panels need to be hidden.

**Comment by samjewell** [comment]
Hi all. I'm an engineer at Grafana, and I too have wanted this feature for ages.

There is something of a workaround / alternative though, which I would like to share. At Grafana we've been developing a new framework for building out Grafana App-plugins, called Scenes. The docs are here: https://grafana.com/developers/scenes/

For me the power of scenes is:
- You can build an app using all the components that are typically available in a dashboard:
   - Variables
   - Queries
   - Panels & Visualisation
   - Transforms
- You assemble it together in Javascript / React, meaning you always have JS as an "escape hatch", if you ever need to build out custom behaviour that isn't possible in Dashboards yet.

Showing or hiding a panel in response to the data is I think one such custom behaviour that could be implemented in JS. 

Check out the `HiddenWhenNoDataBehaviour()` here: https://github.com/grafana/scenes/blob/main/packages/scenes-app/src/demos/behaviors/behaviorsDemo.tsx#L95. We successfully applied this to an app we were building recently.

Scenes is very powerful, and the docs may point you to other tools / app layouts you can use to arrange your content. Having support for Tabs and drills etc is quite exciting. Good luck - I realise this may not help all of you, but I hope it can help some of you.

**Comment by ndjhartman** [comment]
@samjewell , can you comment on @hiepqdinh 's suggested solution? I appreciate the suggestion for scenes, but hiding a panel if the "Repeat For" variable is empty would be a simple solution (for the vast majority of use-cases).

**Comment by guillheu** [comment]
Agreeing with @ndjhartman , If the variable is empty (as in, no values), then have the panel/row be "hidden" (as in, repeated 0 times).

**Comment by samjewell** [comment]
@ndjhartman and @guillheu yes, to me what you're saying makes sense, but although I work at Grafana I'm just a power-user of Grafana at this stage, and not a maintainer who understands what's possible, or the history of past decisions. I believe what you're asking for is hard to do-I think it would actually be a breaking change for many existing dashboards, but I'm not certain of why, or the details, sorry.

**Comment by guillheu** [comment]
> @ndjhartman and @guillheu yes, to me what you're saying makes sense, but although I work at Grafana I'm just a power-user of Grafana at this stage, and not a maintainer who understands what's possible, or the history of past decisions. I believe what you're asking for is hard to do-I think it would actually be a breaking change for many existing dashboards, but I'm not certain of why, or the details, sorry.

I don't know the inner workings of Grafana, but it seems to me that by default, if a variable is "empty" so to speak, it'll take the value "None" instead. I can how changing that behavior would be a breaking change if that's the case. If so, perhaps it's possible to add an option "Enable Empty Variable" or something that's off by default.

**Comment by phyzical** [comment]
@guillheu i had a quick look the other day ( 30 mins) and i think the main reason is there is alot of react hooks floating around so id lean towards yes a major rehaul would be required.

I tried to find a nice place to even perform a "bodge", but from what i could see the "duplicate" logic is more of an afterthought int he component itself, as in it assumes there is always on THEN performs a bunch of logic to duplicate.

Id say if someone designed it from a 0 + concept it would be easy but its just not designed that way atm (i could be wrong)

**Comment by itsjusth** [comment]
Would it be possible to create logic on a row?  Something simple like if $variable=="x" then collapse row and apply css style disabled to the row.  I can't imagine that it would be too difficult to enable something like this.

**Comment by danfoxley** [comment]
"opened
[on Mar 24, 2016](https://github.com/grafana/grafana/issues/4470#issue-143218271)"

**Comment by jounaidr** [comment]
Any update or workaround for this?

**Comment by danfoxley** [comment]
I opened a ticket with Grafana and got this small bit of information:

"this is a request that we have received before and we are considering looking at this next year as part of the dynamic dashboards that the new scenes functionality can support, but for right now this is not something that we are currently progressing."

**Comment by bassedOutt** [comment]
Yes. We have a shared dasbhoard for multiple apps per language stack, since most of the metrics are the same. But some apps are http, other are ws, some have both protocols supported. So it looks weird having empty panels. Would have been nice to hide those if the app does not support the protocol.

**Comment by andreas-ekman-wcar** [comment]
I want this to! Homeassistant has this concept as inspiration.

https://www.home-assistant.io/dashboards/conditional/

**Comment by Gunnnn** [comment]
Useful feature would be!

**Comment by ghost** [comment]
+1

**Comment by vinku34** [comment]
+1

**Comment by sklyfr** [comment]
+1

**Comment by Gunnnn** [comment]
+1

**Comment by noave** [comment]
Would like to show a Panel when there are failed systemd Services, but otherwise I dont want to show 100 green services, or just "nodata" which looks also broken. Of course I can override with "all fine", but why if its best to just hide it..

**Comment by cpdevopsteam** [comment]
10 years later and we still don't have this :p

**Comment by CptOrange16** [comment]
This would be very useful

**Comment by harisrozajac** [comment]
Hello everyone, this is now available as a part of dynamic dashboards, see demo here [Dynamic Dashboards in Grafana 12: Easier to Edit + Responsive | How to | Grafana Labs](https://www.youtube.com/watch?v=tH7dQWRMniM&t=173s). This new experience is currently being rolled out - some of you might have seen it already. We onboard new instances to it every week and aim for general availability for G13.

**Comment by Powerkrieger** [comment]
Is this not available in the open source version?

**Comment by TryTryAgain** [comment]
> Is this not available in the open source version?

I saw a pop up and then following the docs I was able to submit some form for early access (I guess it's still not available by default). I've been waiting for what feels like over a month :/ ...no real way to follow up or see any progress/movement.
